File: SourceAFIS/General/PointF.cs
using System;

namespace SourceAFIS.General
{
    public struct PointF
    {
        public double X;
        public double Y;

        public PointF(double x, double y)
        {
            X = x;
            Y = y;
        }

        public static implicit operator PointF(Point point)
        {
            return new PointF(point.X, point.Y);
        }

        public static PointF operator +(PointF left, SizeF right)
        {
            return new PointF(left.X + right.Width, left.Y + right.Height);
        }

        public static PointF operator *(double factor, PointF point)
        {
            return new PointF(factor * point.X, factor * point.Y);
        }
    }
}

File: java/sourceafis/src/sourceafis/general/PointF.java
package sourceafis.general;

public class PointF {

	public float X;
	public float Y;

	public PointF(float x, float y) {
		X = x;
		Y = y;
	}

	public static PointF toPointF(Point point) {
		return new PointF(point.X, point.Y);
	}

	public static PointF add(PointF left, SizeF right) {
		return new PointF(left.X + right.Width, left.Y + right.Height);
	}

	public static PointF multiply(float factor, PointF point) {
		return new PointF(factor * point.X, factor * point.Y);
	}

	public static java.awt.Point toPoint(PointF point) {
		java.awt.Point p = new java.awt.Point();
		p.setLocation(point.X, point.Y);
		return p;
	}

}
